src/Campaigns/Blocks/CampaignList/app/index.tsx

```diff
@@ -0,0 +1,99 @@
+import {__} from '@wordpress/i18n';
+import {useState} from '@wordpress/element';
+import useCampaigns from '../../shared/hooks/useCampaigns';
+import Pagination from '../../shared/components/Pagination';
+import {CampaignListType} from '../types';
+import {getGoalDescription, getGoalFormattedValue} from '../../CampaignGoal/utils';
+
+import './styles.scss';
+
+const getGridSettings = (layout: string) => {
+    switch (layout) {
+        case 'double':
+            return 2;
+        case 'triple':
+            return 3;
+        default:
+            return 1;
+    }
+}
+
+export default ({attributes}: { attributes: CampaignListType }) => {
+    const [page, setPage] = useState(1);
+
+    const {campaigns, hasResolved, totalPages} = useCampaigns({
+        ids: attributes?.filterBy?.map((item: { value: string }) => Number(item.value)),
+        per_page: attributes?.perPage,
+        sortBy: attributes?.sortBy,
+        orderBy: attributes?.orderBy,
+        page,
+    });
+
+    if (!hasResolved) {
+        return null;
+    }
+
+    return (
+        <>
+            <div
+                className="give-campaigns-campaignListBlock-grid"
+                style={{gridTemplateColumns: `repeat(${getGridSettings(attributes.layout)}, 1fr)`}}
+            >
+                {campaigns?.map((campaign) => (
+                    <div
```

Comment:
I would love to see a Campaign component here 😄 .  That could help separate out the presentational part of the component from the container / logic

Reply:
I agree, that's why this component is coming in next PR https://github.com/impress-org/givewp/tree/feature/campaign-block-GIVE-1388/src/Campaigns/Blocks/shared/components/CampaignCard